crates/types/src/logs/mod.rs

```diff
@@ -387,6 +391,93 @@ impl SequenceNumber for LogletOffset {
     }
 }
 
+mod dto {
+
+    #[derive(Clone, Copy, Debug, bilrost::Oneof, Default)]
+    enum KeyFilterInner {
+        #[default]
+        Any,
+        #[bilrost(1)]
+        Include(u64),
+        // in addition to records with no keys.
+        #[bilrost(2)]
+        Within((u64, u64)),
+    }
+
+    #[derive(Clone, Copy, bilrost::Message)]
+    pub struct KeyFilter {
+        #[bilrost(oneof(1, 2))]
+        inner: KeyFilterInner,
+    }
+
+    impl From<&super::KeyFilter> for KeyFilter {
+        fn from(value: &crate::logs::KeyFilter) -> Self {
+            use crate::logs::KeyFilter::*;
```

Comment:
I wouldn't recommend this unless the number of variants is (a) too large and (b) doesn't include names that might conflict with std lib's prelude types. This case doesn't meet (a) and it's borderline close to hitting (b) due to rust's `Any` type, albeit it's not in prelude.